### DISCUSSION TITLE
Error when building grpc gcc11/ubuntu22.04?

### DISCUSSION THREAD

**Discussion by mgiessing** [triton-inference-server/server#5532] [Q&A]
**Description**
I'm trying to build triton (CPU) r23.02 with ubuntu22.04/gcc-11 and most of the main features are working quite well, except for:
- endpoint=grpc
- filesystem=s3

Therefore I'm having three questions:

1.) Did you already try out to build with gcc11?
2.) Do you already know when you'll bump the base image to ubuntu22.04?
3.) Regarding grpc: When trying to build grpc I'm seeing a lot of `undefined reference to absl::lts_20220623` like seen below:

```bash
[...]
host_port.cc:(.text+0x194): undefined reference to `absl::lts_20220623::string_view::find(char, unsigned long) const'
/usr/bin/ld: host_port.cc:(.text+0x214): undefined reference to `absl::lts_20220623::string_view::find(char, unsigned long) const'
/usr/bin/ld: host_port.cc:(.text+0x28c): undefined reference to `absl::lts_20220623::string_view::find(char, unsigned long) const'
/usr/bin/ld: host_port.cc:(.text+0x31c): undefined reference to `absl::lts_20220623::string_view::find(char, unsigned long) const'
/usr/bin/ld: /tmp/tritonbuild/tritonserver/build/third-party/grpc/lib/libgpr.a(host_port.cc.o): in function `grpc_core::SplitHostPort(absl::lts_20220623::string_view, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::
host_port.cc:(.text+0x42c): undefined reference to `absl::lts_20220623::string_view::find(char, unsigned long) const'
/usr/bin/ld: /tmp/tritonbuild/tritonserver/build/third-party/grpc/lib/libgpr.a(host_port.cc.o):host_port.cc:(.text+0x5f0): more undefined references to `absl::lts_20220623::string_view::find(char, unsigned long) const' follow
/usr/bin/ld: /tmp/tritonbuild/tritonserver/build/third-party/grpc/lib/libgpr.a(host_port.cc.o):(.toc+0x0): undefined reference to `bool absl::lts_20220623::str_format_internal::FormatArgImpl::Dispatch<absl::lts_20220623::string_view>(absl::lts_20220623:'
collect2: error: ld returned 1 exit status
```

**Triton Information**
r23.02

Are you using the Triton container or did you build it yourself?
Build it myself

**To Reproduce**

Clone core repository and changed the following to overcome an error that was probably caused by gcc-11:

```bash
sed -i "s/Werror/Werror -Wno-error=range-loop-construct -Wno-error=maybe-uninitialized/g" src/CMakeLists.txt
```

Clone the server repository and build like this:

```bash
./build.py \
  --image=base,docker.io/ubuntu:22.04 \
  --github-organization https://github.com/mgiessing \
  --build-id $(date +"%d%m%Y") \
  --repo-tag=core:ppc64le_${VER} \
  --endpoint=grpc
```

**Expected behavior**
It should build without errors :)

**Comment by the-david-oy** [comment]
Moving this to Discussions, because we [do not officially support Ubuntu 22.04](https://github.com/triton-inference-server/server/blob/main/docs/customization_guide/build.md#building-triton) yet. The build guide does have a section on building [on unsupported platforms](https://github.com/triton-inference-server/server/blob/main/docs/customization_guide/build.md#building-on-unsupported-platforms).

We haven't yet released a public date for moving to 22.04, at which time the containers would be updated and tested. You may be able to get community support here from others who have built for 22.04.

Otherwise, you may want to try using our pre-built containers or building in a 20.04 container. Your OS does not need to match the version of Ubuntu used in the container.
